mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/KeyValueBasedPersistenceProvider.java

```diff
@@ -421,23 +421,23 @@ public List<IJobClusterMetadata> loadAllJobClusters() throws IOException {
     public List<CompletedJob> loadAllCompletedJobs() throws IOException {
         AtomicInteger failedCount = new AtomicInteger();
         AtomicInteger successCount = new AtomicInteger();
-
-        final List<CompletedJob> completedJobsList = Lists.newArrayList();
-        final String namespace = NAMED_COMPLETEDJOBS_NS;
-        for (String pkey : kvStore.getAllPartitionKeys(namespace)) {
-            kvStore.getAll(namespace, pkey).values()
-                .forEach(data -> {
-                    try {
-                        NamedJob.CompletedJob cj = mapper.readValue(data, NamedJob.CompletedJob.class);
-                        CompletedJob completedJob = DataFormatAdapter.convertNamedJobCompletedJobToCompletedJob(cj);
-                        completedJobsList.add(completedJob);
-                        successCount.getAndIncrement();
-                    } catch (JsonProcessingException e) {
-                        logger.warn("failed to parse CompletedJob from {} {}", pkey, data, e);
-                        failedCount.getAndIncrement();
-                    }
-                });
-        }
+        final List<CompletedJob> completedJobsList = kvStore.getAllRows(NAMED_COMPLETEDJOBS_NS)
+            .values().stream()
+            .flatMap(x -> x.values().stream())
+            .map(data -> {
+                try {
+                    NamedJob.CompletedJob cj = mapper.readValue(data, NamedJob.CompletedJob.class);
+                    CompletedJob completedJob = DataFormatAdapter.convertNamedJobCompletedJobToCompletedJob(cj);
+                    successCount.getAndIncrement();
+                    return completedJob;
+                } catch (JsonProcessingException e) {
+                    logger.warn("failed to parse CompletedJob from {}", data, e);
```

Comment:
Do we expect this to happen in any case? if this error does happen does it imply some regression and would it be better to fail loudly instead?

Reply:
it implies some regression but since completed jobs aren't critical for functionality I'm okay with this as a warning instead.